lib/vdsm/storage/mpathconf.py

```diff
@@ -130,3 +385,46 @@ def read_blacklist():
         wwids.add(wwid)
 
     return wwids
+
+
+def configure_multipathd():
+    """
+    Set up multipath daemon configuration to the known and supported state.
+    The original configuration, if any, is saved.
+
+    Returns:
+        str: The path to the copy of the old multipath.conf. Can be empty.
+    """
+    backup = fileUtils.backup_file(CONF_FILE)
+
+    data = _CONF_DATA.encode('utf-8')
+    fileUtils.atomic_write(CONF_FILE, data, relabel=True)
+    return backup
+
+
+def read_metadata():
+    """
+    The multipath configuration file at /etc/multipath.conf should contain
+    a tag in form "RHEV REVISION X.Y" for this check to succeed.
+    If the tag above is followed by tag "RHEV PRIVATE" the configuration
+    should be preserved at all cost.
+
+    Returns:
+        Metadata: Return a tuple where the first element is the revision
+            check result, and the second is private check result.
+    """
+    with open(CONF_FILE) as f:
+        # Read only the lines that are going to be processed
+        first = f.readline().rstrip("\n")
+        second = f.readline().rstrip("\n")
+
+    private = _PRIVATE_TAG in second or _OLD_PRIVATE_TAG in second
+
+    if CURRENT_TAG in first:
+        return Metadata(REVISION_OK, private)
+
+    for tag in _OLD_TAGS:
+        if tag in first:
+            return Metadata(REVISION_OLD, private)
```

Comment:
Just because I like simple one-liners, this could also be written as:
```python
    if any(tag in first for tag in _OLD_TAGS):
        return Metadata(REVISION_OLD, private)
```
Both should have the same cost, and for this simple loop it is not complicated to read.

Honestly, I don't think I will change it, not worth it. But I thought it might be interesting to know your opinion on this style @nirs :)

Reply:
Using any is nicer, and the cost does not matter sonce we run this once per install/upgrade.

But can we use this?

    first.startswith(_OLD_TAGS)

Reply:
Oh! I didn't know `startswith` could take an iterable too!

Tried it locally, and it passes the test, so it seems to work correctly :+1: 

I can push it, then, and the patch would be ready to review. No need to run OST again, the code in this patch is not covered atm.